sdk/src/Core/Amazon.Util/Internal/JsonSerializerHelper.cs

```diff
@@ -29,6 +48,53 @@ public partial class EC2InstanceMetadataJsonSerializerContexts : JsonSerializerC
     {
     }
 
+    [JsonSerializable(typeof(Dictionary<string, string>))]
+    public partial class DictionaryStringStringJsonSerializerContexts : JsonSerializerContext
+    {
+#if !NET6_0_OR_GREATER
+        public DictionaryStringStringJsonSerializerContexts(JsonSerializerOptions defaultOptions)
+            : base(defaultOptions)
+        {
+        }
+#endif
+    }
+
+    [JsonSerializable(typeof(Dictionary<string, Dictionary<string, string>>))]
+    public partial class DictionaryStringDictionaryStringJsonSerializerContexts : JsonSerializerContext
+    {
+#if !NET6_0_OR_GREATER
+        public DictionaryStringDictionaryStringJsonSerializerContexts(JsonSerializerOptions defaultOptions)
+            : base(defaultOptions)
+        {
+        }
+#endif
+    }
+
+    [JsonSerializable(typeof(List<string>))]
+    public partial class ListStringJsonSerializerContexts : JsonSerializerContext
+    {
```

Comment:
Any reason why only the `DictionaryString*` classes have a constructor with `JsonSerializerOptions`? Are those not required for the other contexts?

---

A separate question: Are we going to have to create context classes for every type that'll be serialized?

Reply:
It was the only one used with JSON pretty print or `WriteIndented` used when serializing. Most of our JSON use in Core is actually deserialization so there is no need for pretty print. If we find we need more with pretty print we can add constructors.

For your second question yes we always need a context class so we can have the source generated JSON marshalling code. I debated about having one context with a lot of `JsonSerializable` for each of the types. I tried something similar for the Lambda repository and ran into issues when the types referenced in the `JsonSerializable` attribute had overlapping subtypes which caused code generation issues.